subsys/net/lib/nrf_cloud/src/nrf_cloud_transport.c

```diff
@@ -221,6 +229,39 @@ static uint32_t dc_send(const struct nct_dc_data *dc_data, uint8_t qos)
 	return mqtt_publish(&nct.client, &publish);
 }
 
+static int batch_send(const struct nct_dc_data *dc_data, enum mqtt_qos qos)
+{
+	if (dc_data == NULL) {
+		LOG_DBG("Passed in structure cannot be NULL");
+		return -EINVAL;
+	}
+
+	if (qos != MQTT_QOS_0_AT_MOST_ONCE && qos != MQTT_QOS_1_AT_LEAST_ONCE) {
+		LOG_DBG("Unsupported MQTT QoS level");
+		return -EINVAL;
+	}
+
+	struct mqtt_publish_param publish = {
+		.message.topic.qos = qos,
+		.message.topic.topic.size = nct.dc_batch_endp.size,
+		.message.topic.topic.utf8 = nct.dc_batch_endp.utf8,
+	};
+
+	/* Populate payload. */
+	if ((dc_data->data.len != 0) && (dc_data->data.ptr != NULL)) {
+		publish.message.payload.data = (uint8_t *)dc_data->data.ptr;
+		publish.message.payload.len = dc_data->data.len;
+	} else {
+		LOG_DBG("Payload is empty!");
```

Comment:
This problem was preexisting, but, is it valid to simply pass an empty payload to `mqtt_publish()`?  The docs for it say "Parameters to be used for the publish message. Shall not be NULL."

Reply:
this is fine, the parameters `&nct.client` and `&publish` are not NULL.